src/ocean/task/internal/SpecializedPools.d

```diff
@@ -0,0 +1,124 @@
+/*******************************************************************************
+
+    Copyright: Copyright (c) 2017 sociomantic Labs GmbH. All rights reserved.
+
+    License:
+        Boost Software License Version 1.0. See LICENSE_BOOST.txt for details.
+        Alternatively, this file may be distributed under the terms of the Tango
+        3-Clause BSD License (see LICENSE_BSD.txt for details).
+
+*******************************************************************************/
+
+module ocean.task.internal.SpecializedPools;
+
+import ocean.task.internal.FiberPoolEager;
+import ocean.task.Task;
+import ocean.core.Verify;
+import ocean.core.Enforce;
+
+/*******************************************************************************
+
+    Defines single mapping of `ClassInfo` to worker fiber pool in configuration
+
+*******************************************************************************/
+
+public struct PoolDescription
+{
+    /// result of `Task.classinfo` for task type which is to be handled
+    /// by this pool
+    ClassInfo task_kind;
+
+    /// worker fiber allocate stack size
+    size_t stack_size;
+}
+
+/*******************************************************************************
+
+    Defines all mappings between task kinds and dedicated worker fiber pools
+    processing those.
+
+*******************************************************************************/
+
+public class SpecializedPools
+{
+    /// mapping as a simple array (expected to contain few elements)
+    private SpecializedPool[] mapping;
```

Comment:
What's the reason you're not using an AA?

Reply:
Cons: slower, inherent heap fragmentation, have to keep in mind all const-related AA bugs
Pros: nice syntax

I think for purely private entity in stdlib choice is obvious :P